helm/mongodb/templates/pv.yaml

```diff
@@ -6,21 +10,23 @@ metadata:
     pv.kubernetes.io/bound-by-controller: "yes"
     pv.kubernetes.io/provisioned-by: kubernetes.io/aws-ebs
   labels:
-    failure-domain.beta.kubernetes.io/region: {{ .Values.pv0.region }} 
-    failure-domain.beta.kubernetes.io/zone: {{ .Values.pv0.zone }} 
-  name: {{ .Values.pv0.name }} 
+    failure-domain.beta.kubernetes.io/region: {{ $pv.region }}
+    failure-domain.beta.kubernetes.io/zone: {{ $pv.zone }}
+    provisioned-for: {{ $values.service.name }}-{{ $index }}
+  name: {{ $pv.name }}
   namespace: "default"
 spec:
   accessModes:
   - ReadWriteOnce
   awsElasticBlockStore:
-    fsType: {{ .Values.fsType }} 
-    volumeID: {{ .Values.pv0.volumeID }} 
+    fsType: {{ $values.fsType }}
+    volumeID: {{ $pv.volumeID }}
   capacity:
-    storage: {{ .Values.storageGi }} 
+    storage: {{ $values.storageGi }}
   claimRef:
```

Comment:
We don't need this claim ref any more. I think this was needed when we used stateful sets.
The matching is done based on the labels.